content/posts/scientific-python/2023-developer-summit/index.md

```diff
@@ -0,0 +1,91 @@
+Some big picture take-aways are:
+
+- We reorganized how to check for matrix/array/format info. This involved
+  adding a `format` attribute describing which format of sparse storage is used,
+  adding an `_is_array` attribute to assist in converting from the matrix to array
+  interfaces, changing functions `issparse`/`isspmatrix` as well as shifting
+  the class hierarchy to allow easy `isinstance` checking.
+  The interface going forward includes:
+  - `issparse(A)`: True when a sprase array or matrix.
+  - `isinstance(A, sparray)` or alternatively `issparse(A) and A._is_array`: True when a sprase array.
+  - `isspmatrix(A)` or alternatively `issparse() and not A._is_array`: True when a sparse matrix.
+    To check the format of a sparse array or matrix use `A.format == "csr"` or similar.
+- We updated the `sparray` interface, leaving `spmatrix` untouched for backward compatibility.
+- We made decisions about how to approach the "creation functions" for sparse arrays.
+  The big-picture approach is to introduce new functions with an `_array` suffix which
+  construct sparse arrays. The old names will continue to create sparse matrix until
+  post-deprecation removal.
+  Some specific changes made include:
+  - Add the creation function `diags_array(A)` (and planned for `eye_array`, `random_array` and others).
+  - Create a `sparse.linalg.matrix_power` function for positive integer matrix power of a sparse array
+- Made progress toward 1D sparse arrays. The data structures for 1d may be quite different from 2d.
+  A prototype `coo_array` allowed exploration of possible n-d arrays, though that is not a short-term goal.
+- Explored feasibility and usefulness of defining `__array_ufunc__` and other `__array_*__` protocols for sparse arrays
+- Improved documentation for sparse arrays
+
+We have set a goal of a working set of sparse array construction functions
+and a 1d sparse array class (focusing on `coo_array` first) in plenty of
+time for intensive testing before SciPy v1.12. This will then allow us to
+focus on creating migration documents and tools as well as helping downstream
+libraries make the shift to sparse arrays. We hope to enable the removal of
+deprecated sparse matrix interfaces in favor of the array interface. For this
+to happen we will need most downstream users to shift to the sparse array api.
+We intend to help them do that.
+
+Our work continues with a community call every [two weeks on Fridays.](https://scientific-python.org/calendars)
+Our immediate near term work involves:
+
+- Continue improving sparse creation functions: diags, eye, random and others.
+- Deprecate some matrix-specific functionality
+- General performance improvements
+- Adapting scikit-learn to support sparse arrays (to be discussed with scikit-learn's maintainers)
+
+This blog was largely composed based on the [scipy-sparse report hackmd document](https://hackmd.io/1Q2832LDR_2Uv_-cV-wnYg).
```

Comment:
This document is mentioned earlier.

```suggestion
```

Reply:
Yes -- I was giving credit here. And didn't remove it when I added the link elsewhere. This can be deleted.